### PR TITLE
executing notebooks w/ nbclient in their directory

### LINKED ISSUE #180: nbclient execution doesn't do so in the notebook's directory
In https://github.com/executablebooks/jupyter-book/issues/605 @bmcfee mentioned that notebooks that have relative dependencies in the code weren't working. I think that's be cause when we execute with `nbclient` (e.g. if execute == `auto`) then we don't execute the notebook in the directory where it exists: https://github.com/executablebooks/MyST-NB/blob/bb4f754c328f7d0e3fa6b6a18a93eaa0f5befacf/myst_nb/cache.py#L147

I think this can be fixed by specifying a `cwd=` argument to `execute(` (https://github.com/jupyter/nbclient/blob/2d862a9ab6f34c6c93761283aaa42d8f39dcf8e2/nbclient/client.py#L858) that points to the folder where the notebook lives

### PR DESCRIPTION
**Pull request by choldgraf**
This adds the `cwd` command to our call to `nbclient.execute`, and also updates a test to use a relative file path in the code to make sure it works

fixes #180

### CODE REVIEW

**Comment by choldgraf** [comment]
@AakashGfude does this look correct to you?

**Comment by AakashGfude** [comment]
> @AakashGfude does this look correct to you?

@choldgraf Sorry missed this one. One question though, should we be asserting for "Execution succeeded" instead of "Executing" in the tests, to make sure everything is running fine? as in `cache.py` the info "Executing" is run before the execute command. https://github.com/executablebooks/MyST-NB/blob/master/myst_nb/cache.py#L148

**Comment by choldgraf** [comment]
@AakashGfude yep we should, I guess I missed that...wanna open an issue about it? (or a PR?)

**Comment by AakashGfude** [comment]
@choldgraf sure will open a PR tomorrow morning

File: tests/notebooks/basic_relative.ipynb [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Test reading a relative file to make sure relative paths work\n",
+    "with open(\"./conf.py\") as ff:\n",
```
Comment (AakashGfude) [inline tests/notebooks/basic_relative.ipynb:19]: @choldgraf I guess this PR is missing the `conf.py` file?

Comment (choldgraf) [reply tests/notebooks/basic_relative.ipynb:19]: no it shows up when the test is run - I think the fixture builds a little sphinx site in a tmp folder, so `conf.py` is there

Comment (AakashGfude) [reply tests/notebooks/basic_relative.ipynb:19]: @choldgraf ooh cool. gotcha.

Comment (AakashGfude) [reply tests/notebooks/basic_relative.ipynb:19]: @choldgraf but it is showing the error of file not found for the test in the local system.

sorry for the unclear screenshot. I had put a debugger in `test_relative_path_cache`

image

So, the execution failed, and when I saw the log file mentioned in the screenshot. It said `conf.py` file not found. Can you verify it in your system?

Comment (choldgraf) [reply tests/notebooks/basic_relative.ipynb:19]: hmmm, that is weird - I just tried it and it worked for me. are you on latest myst-nb/jupyter-cache?

Comment (AakashGfude) [reply tests/notebooks/basic_relative.ipynb:19]: Hmm. this is strange. Like for example, I created the PR, in which I changed the assert statement to `Execution Succeeded`. https://github.com/executablebooks/MyST-NB/pull/199 . And it is failing, similar to in my local system.

**Pull request merged**